app/frontend/wallet/cardano-wallet.js

```diff
@@ -165,12 +165,24 @@ const CardanoWallet = async (options) => {
       txInputs.push(TxInputFromUtxo(utxos[i]))
       sumUtxos += utxos[i].coins
 
-      totalCoins = coins + computeTxFee(txInputs, totalCoins, utxos[i].address)
+      totalCoins = coins + computeTxFee(txInputs, totalCoins, address)
     }
 
     return txInputs
   }
 
+  async function prepareTxInputsOrFail(coins, address) {
+    const txInputs = await prepareTxInputs(coins, address)
+    const txInputsCoinsSum = txInputs.reduce((acc, elem) => {
```

Comment:
this can be simplified to
`txInputs.reduce((acc, elem) => acc + elem.coins, 0) `